Add rectangular JupyterHub logo

rgaiacs opened the pull request:
Related to https://github.com/jupyterhub/design/issues/9

- [x] re-organise folder structure by splitting design / guide and assets
- [x] document design / guide on `README`
- [x] add rectangular JupyterHub logo

  This is a re-construction based on [Project Jupyter Branding Guidelines](https://github.com/jupyter/design/blob/main/brandguide/brand_guide.pdf) to have the text editable.
- [x] add rectangular Binder logo

  This is a re-construction based on the existing design and using the same font used in JupyterHub logo.
- [x] add square JupyterHub logo
- [x] add square Binder logo

rgaiacs committed:
Re-work the logos

manics commented:
assets/jupyterhub/jupyterhub-logo-square-color.png is an svg with a png extension.

The SVGs reference a font which isn't installed by default. Can you convert the text to a path instead?

rgaiacs committed:
Add pre-commit configuration

rgaiacs committed:
Add Python script to build assets

rgaiacs commented:
> The SVGs reference a font which isn't installed by default. Can you convert the text to a path instead?

This pull request now includes a directory `src` with the SVG containing letters as text and a directory `assets` with the SVG containing letters as path.

The content of the directory `assets` is generated using a small Python script to call Inkscape. The Python script can have some improvements in future pull requests, for example

- generate PNG with different sizes
- generate SVG with the same height
- generate SVG and PNG with single color

manics commented on .flake8:
```diff
@@ -0,0 +1,16 @@
+
+# Adjustments to linting the binderhub repo
+builtins = c, load_subconfig
+exclude = versioneer.py,binderhub/_version.py
```
```suggestion
```

manics commented on README.md:
```diff
@@ -1,19 +1,70 @@
+### Binder
 
-JupyterHub logos
+Assets of the [Binder] project is located in the [`assets/binder`](./assets/binder) directory.
```
```suggestion
Assets of the [Binder] project are located in the [`assets/binder`](./assets/binder) directory.
```

manics commented on README.md:
```diff
@@ -1,19 +1,70 @@
+### JupyterHub
 
-## Project Jupyter Design Assets
+Assets of the [JupyterHub] project is located in the [`src/jupyterhub`](./src/jupyterhub) directory.
```
```suggestion
Assets of the [JupyterHub] project are located in the [`src/jupyterhub`](./src/jupyterhub) directory.
```

manics commented on README.md:
```diff
@@ -1,19 +1,70 @@
-[jupyter/design](https://github.com/jupyter/design) repo.
+## Source
+
+The `.svg` files are provided for reproducibility. This files **must** only be consumed by [`build-assets.py`](./build-assets.py).
```
```suggestion
The `.svg` files in this directory are provided for reproducibility. This files **must** only be consumed by [`build-assets.py`](./build-assets.py).
You must install the following before running the script:
- [Inkscape](https://inkscape.org/) must be in your PATH.
- The [Myriad Pro](link to the actual font that can be downloaded) font
```
For reproducibility do you have a link to the exact font you used? There's no guarantee that a Google result will give the official font.

rgaiacs committed:
Clean up flake8 configuration

Based on review from @manics.

rgaiacs committed:
Fix grammar in README

Based on review from @manics.

rgaiacs committed:
Add source for the font

to resolve comment from @manics.

choldgraf reviewed (approved):
Overall this looks like a really nice improvement, I like the re-organization and the extra context you've added to help others understand the repository.

My main concern is that the fonts all feel off - I think it's because they're non-standard fonts as @manics mentioned. I suggest that we:

- Use high-res PNGs of the in-production logos if we can't easily find the font
- Merge this PR
- Open an issue about the need to pick a similar font that isn't proprietary / custom for our logos
- Track the logo font upgrades via that issue

manics commented:
I think the fonts are off because the text hasn't been converted to a path for all assets. It should be possible to fix.

choldgraf commented:
@manics ahhh I understand what you mean by path now, you're basically saying "turn the text into shapes rather than text". Makes sense to me and i agree that's better than having them as PNGs.

FWIW, I've been maintaining this Figma board with a bunch of Jupyter assets for a few years now. Would it help to make something similar for jupyterhub, or just re-use that one at all?

https://www.figma.com/design/20E31Z8TV1HLvDPn3ckTRR/Jupyter-Design-Assets?node-id=0-1&t=pQ0xyYWo4VCUGos4-1

rgaiacs committed:
Fix missing call to svg2svg() in build-assets.py

rgaiacs committed:
Fix SVG assets to use path

rgaiacs commented:
@manics thanks for catching the missing call `svg2svg()` in the script. I fixed it and updated the assets. Below is a side by side comparison. The screenshot is on a machine that does **not** have the font.

<img width="906" height="97" alt="image" src="https://github.com/user-attachments/assets/627c30a3-12c2-4736-abc7-d98e93140406" />

The source is uses a fall back font. The SVG and PNG looks like with the correct font.

rgaiacs commented:
> Would it help to make something similar for jupyterhub, or just re-use that one at all?

I have zero experience with Figma. The reasons that I like about having the assets on GitHub are

- it is under version control, i.e. it is easy to tell when things change
- it allows automation of downstream projects, i.e. ReadTheDocs can query this repository for the correct logo, e.g. https://raw.githubusercontent.com/jupyterhub/design/refs/heads/master/jupyterhub/jupyterhub-squarelogo-bare.svg.
- it has a place to ask questions and contribute back

choldgraf commented:
Oh I'm definitely not suggesting we replace this repo with figma, just offering to document it as a resource, not a source of truth (and i think of this repo as a source of truth)

rgaiacs commented:
> just offering to document it as a resource

Looks good to me.
